crates/core/src/replication/parser.rs

```diff
@@ -0,0 +1,291 @@
+use crate::{Error, Result};
+use fallible_iterator::FallibleIterator;
+use sqlite3_parser::ast::{Cmd, PragmaBody, QualifiedName, Stmt};
+use sqlite3_parser::lexer::sql::{Parser, ParserError};
+
+/// A group of statements to be executed together.
+#[derive(Debug, Clone)]
+pub struct Statement {
+    pub stmt: String,
+    pub kind: StmtKind,
+    /// Is the statement an INSERT, UPDATE or DELETE?
+    pub is_iud: bool,
+    pub is_insert: bool,
```

Comment:
let's just reuse the`Stmt` enum, with `Stmt::Insert` etc? Instead of `is_iud` and `is_insert`, they have combinations that don't make (`is_iud = false && is_insert = true`)

Reply:
done